### ISSUE TITLE
jax svd is exceptionally slow on TPUs

### ISSUE THREAD

**Issue by rgreenblatt** [google/jax#10052]
The runtime is much slower than on cpu and also slower than `tf`.

```python
from time import time
from functools import partial

import jax
import jax.numpy as jnp


run_svd = jax.jit(partial(jnp.linalg.svd, full_matrices=False))

mat = jax.random.normal(jax.random.PRNGKey(2), (128, 64))

jax.block_until_ready(run_svd(mat))

start = time()
jax.block_until_ready(run_svd(mat))
print(time() - start)
```

For me, this prints out around ~0.35 seconds on a tpu v-3.

This tf code takes around 0.0015 which seems much more reasonable:
```
import tensorflow as tf

tf_mat = tf.random.normal((128, 64))

tf.linalg.svd(tf_mat, full_matrices=False)

start = time()
s, u, v = tf.linalg.svd(tf_mat, full_matrices=False)
print(time() - start)
```

The `hermitian=True` case is fine.

**Comment by hawkinsp** [comment]
Yes, this is a known issue. The TPU SVD implementation is quite naive at the moment: it's a non-parallel Jacobi solver that performs terribly on accelerators. Symmetric eigendecomposition (the `hermitian` case) is much more optimized, especially for small matrices.

There's an open PR (https://github.com/google/jax/pull/9681) that replaces it with an approach based on the polar decomposition that hopefully we should be able to merge soon.

**Comment by tlu7** [comment]
The PR on qdwh-svd solver (https://github.com/google/jax/pull/10473) is ready to merge. Hope this will address your concern on TPU performance.
